nsot/models/resource.py

```diff
@@ -69,40 +77,73 @@ def set_query(self, query, site_id=None, unique=False):
                 log.debug('Regex enabled for %r' % name)
 
             # Attribute lookup params
-            params = dict(
-                name=name, resource_name=resource_name
-            )
+            if name not in concrete_field_names:
+                params = dict(
+                    name=name, resource_name=resource_name
+                )
+            else:
+                params = dict()
+                params[name] = value
             # Only include site_id if it's set
             if site_id is not None:
                 params['site_id'] = site_id
 
             # If an Attribute doesn't exist, the set query is invalid. Return
             # an empty queryset. (fix #99)
-            try:
-                attr = Attribute.objects.get(
-                    **params
-                )
-            except Attribute.DoesNotExist as err:
-                raise exc.ValidationError({
-                    'query': '%s: %r' % (err.message.rstrip('.'), name)
-                })
+            if name in concrete_field_names:
+                try:
+                    # import pdb; pdb.set_trace()
+                    qfield = self.model.objects.filter(
+                        **params
+                    )
+                except self.model.DoesNotExist as err:
+                    raise exc.ValidationError({
+                        'query': '%s: %r' % (err.message.rstrip('.'), name)
+                    })
+
+            else:
+                try:
+                    qfield = Attribute.objects.get(
+                        **params
+                    )
+                except Attribute.DoesNotExist as err:
+                    # If the query field is not an attribute, check if it is a
+                    # concrete field on the resource. If not, then raise
+                    # an exception
+                    raise exc.ValidationError({
+                        'query': '%s: %r' % (err.message.rstrip('.'), name)
+                    })
 
             # Set lookup params
-            next_set_params = {
-                'name': attr.name,
-                'value': value,
-                'resource_name': resource_name
-            }
+            if name not in concrete_field_names:
+                next_set_params = {
+                    'name': qfield.name,
+                    'value': value,
+                    'resource_name': resource_name
+                }
+            else:
+                next_set_params = {
+                    name: value
+                }
 
             # If it's a regex query, swap ``value`` with ``value__regex``.
             if regex_query:
                 next_set_params['value__regex'] = next_set_params.pop('value')
 
-            next_set = Q(
-                id__in=Value.objects.filter(
-                    **next_set_params
-                ).values_list('resource_id', flat=True)
-            )
+            # if name in concrete_field_names:
```

Comment:
Can you please replace this comment with one that explains how `qfield` is used here to filter out non-concrete-fields.

Reply:
Done.